test/test-lex.cpp

```diff
@@ -161,22 +161,21 @@ TEST(test_lex, lex_binary_numbers) {
   check_single_token(u8"0B010101010101010", token_type::number);
 }
 
-// TODO(mc2) binary numbers may not have decimals
-TEST(test_lex, DISABLED_fail_lex_binary_number) {
-  {
-    error_collector v;
-    padded_string input(u8"0b1.1");
-    lexer l(&input, &v);
-    EXPECT_EQ(l.peek().type, token_type::number);
-    l.skip();
-    EXPECT_EQ(l.peek().type, token_type::end_of_file);
-
-    // q(🎅🏾) have another error kind for
-    // `unexpected_character_in_binary_number?
-    EXPECT_THAT(v.errors, ElementsAre(ERROR_TYPE_FIELD(
-                              error_unexpected_characters_in_number, characters,
-                              offsets_matcher(&input, 3, 4))));
-  }
+TEST(test_lex, fail_lex_binary_number_no_digits) {
+  check_tokens_with_errors(
+      u8"0b", {token_type::number},
+      [](padded_string_view input, const auto& errors) {
+        EXPECT_THAT(errors, ElementsAre(ERROR_TYPE_FIELD(
+                                error_no_digits_in_binary_number,
+                                characters, offsets_matcher(input, 2, 2))));
```

Comment:
Blocking: Shouldn't we highlight the `0b`? As written, the end of the file is highlighted.
```suggestion
                                characters, offsets_matcher(input, 0, 2))));
```

Reply:
this is resolved in commit 22c093d81ac5b544d3b4f92e07ecf2fe1423b3dd

Reply:
I don't have to pass the number literal prefix pointer to parse functions because there's [`lex::last_token_.begin`](https://github.com/catac1/quick-lint-js/blob/1538d2a80a24032d71d70e17a35e9d2852acbf25/src/quick-lint-js/lex.h#L365) 
member variable and it is set to prefix location before all litenal number parse function calls at here [lex.cpp#L150](https://github.com/catac1/quick-lint-js/blob/1538d2a80a24032d71d70e17a35e9d2852acbf25/src/lex.cpp#L150)